File: robot_examples/go/rest_api/swagger/RobotCommand.go
package swagger

import (
)

type RobotCommand struct {
    Token  string  `json:"token,omitempty"`
    Fire  FireCommand  `json:"fire,omitempty"`
    Drive  DriveCommand  `json:"drive,omitempty"`
    Scan  ScanCommand  `json:"scan,omitempty"`
    
}

File: client/generated_rest_api/haskell-servant/lib/Model/RobotCommand.hs
{-# LANGUAGE DataKinds #-}
{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE TypeOperators #-}
{-# LANGUAGE OverloadedStrings #-}
{-# LANGUAGE GeneralizedNewtypeDeriving #-}

module Model.RobotCommand
    ( RobotCommand (..)
    ) where

import Data.Aeson
import GHC.Generics
import Test.QuickCheck
import Model.DriveCommand
import Model.FireCommand
import Model.ScanCommand


data RobotCommand = RobotCommand
    { token :: String
    , fire :: FireCommand
    , drive :: DriveCommand
    , scan :: ScanCommand
    } deriving (Show, Eq, Generic)

instance FromJSON RobotCommand
instance ToJSON RobotCommand
instance Arbitrary RobotCommand where
    arbitrary = RobotCommand <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
